include/aikido/trajectory/util.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef AIKIDO_TRAJECTORY_UTIL_HPP_
+#define AIKIDO_TRAJECTORY_UTIL_HPP_
+
+#include <aikido/trajectory/Interpolated.hpp>
+#include <aikido/trajectory/Spline.hpp>
+
+namespace aikido {
+namespace trajectory {
+
+/// Convert an interpolated trajectory to a piecewise linear spline trajectory
+/// This function requires the \c _inputTrajectory to use a \c
+/// GeodesicInterpolator.
+/// So the conversion exactly preserves the geometric path
+/// \param[in] _inputTrajectory interpolated trajectory
+/// \return a spline trajectory
+std::unique_ptr<aikido::trajectory::Spline> convertToSpline(
+    const aikido::trajectory::Interpolated& _inputTrajectory);
+
+/// Convert a piecewise linear spline trajectory to an interpolated trajectory
+/// using a given interpolator.
+/// \param[in] spline trajectory
+/// \param[in] interpolator used in connecting two ends of a segment
+/// \return an interpolated trajectory
+std::unique_ptr<aikido::trajectory::Interpolated> convertToInterpolated(
+    const aikido::trajectory::Spline& traj,
+    const aikido::statespace::InterpolatorPtr interpolator);
+
+/// Concatenate two spline trajectories
+/// This function converts two spline trajectories into interpolated
+/// trajectories
+/// (geodesic interpolator) for concatenation, then convert the concatenated
+/// interpolated to spline. The start state of the last segment in the first
+/// trajectory is connected with the start of the first segment in the second
+/// trajectory in concatenation.
```

Comment:
Not in this implementation. We do not enforce the match. Because in most of the cases, the end of the first does not match the start of the second. Maybe I should not name it `concatenate`. Do you think `merge` is better? Instead of connecting the end of the last segment with the start of the second, it directly connects the start of the last segment.

Reply:
So we concatenate the two (and interpolate), with the time interval between the two points the same as `dt`? Then I think it may be nice to have a tolerance param to check for maximum distance between the two points so that two widely different points do not concatenate. I'm fine with the current name as long as it's explained clearly.

Reply:
Looking at the implementation, it seems the last waypoint of the first trajectory is ignored and the first waypoint of the second trajectory is used at the end time of the first trajectory.

For example, the following trajecoty
```
traj1: wp1(t=0), wp2(t=1.1), wp3(t=3.0)
traj2: wp4(t=0), wp5(t=5.0)
```
becomes
```
new_traj: wp1(t=0), wp2(t=1.1), wp4(t=3.0), wp5(t=8.0).
```
so that the total time of the new trajectory is the sum of the durations of the first and second trajectories.

We may want to be able to set the policy of which waypoint is used for the "junction waypoint", but the current implementation would be okay as long as it's clearly documented.

Reply:
Yes. It is exactly like what @jslee02 described. I will use this example to explain.